### PR TITLE
Creating Posts and Users - milestone 3

### PR DESCRIPTION
**Pull request by adaorachi**
This is the first draft of the milestone 3 of this project.

The following features were implemented.

- Created session authentication using Devise gem

- Created the Users controller

- Created the Post model and controller

- Implemented unit testing using Rspec & Capybara

Kindly review.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @posts = Post.all
+
+    
+  end
+
+  def create
+    
+
+    @post = Post.new
+    @posts = Post.all
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Not needed.
According to your UI, you can create a form on Post#index view and access to that method only when the user submit the post creation form

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+   
+
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to posts_path
+    else
+      render :index
+    end
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Don't forget to send errors to the view

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+class PostsController < ApplicationController
+  def index
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: You must add a `before_action` statement to ensure that only signed-in user can access to that controller

**Comment by samgaco** [comment]
Hi Team, great work on the design :+1: 

All the previous requirements have been addressed. You can move on now :+1: 

----------------------

* ###### STATUS OF THE MILESTONE: APPROVED :heavy_check_mark:

**Pull request merged**